sympy/physics/quantum/lapply.py

```diff
@@ -0,0 +1,1419 @@
+        >>> lapply(cp.subs(x, q), Add=quatAdd)
+        0 + 0*i + 0*j + 0*k
+
+        If real quaternions should be identified as commutative add a handler
+        for Quaternions to the c_nc_ncef Dispatcher of lapply:
+
+        >>> c_nc_ncef.add((Quaternion,), (lambda q, to_L, **options: \
+             ([q.scalar_part()], [], S.One) if q.vector_part().is_zero_quaternion() \
+                        else ([],[], q) ))
+        >>> lapply(q * q.conjugate())
+        a**2 + b**2 + c**2 + d**2
+    """
+    # if e is no SymPy Expr, return it unmodified (e.g. Numbers).
+    # It is useless to call lapply with Basic or even less object types, as
+    # the purpose of lapply requires that at least the arithmetic operation
+    # "apply" is defined on all objects in e in some form. Since in SymPy
+    # "Everything that requires arithmetic operations to be defined should
+    # subclass Expr" (see definition of class Expr), and since this holds
+    # for numbers and symbols too, no need to consider type Basic anywhere.
+    if not isinstance(e, Expr) or e.is_Atom:
+        return e
+
+    # set options for lapply_* and helper functions:
+    # distribute commutative factors and apply hint 'mul' of expand()
+    options['mul'          ] = opt_mul = mul
+    # for commutative factors: apply hint 'power_base' of expand()
+    if power_base    is None: power_base    = opt_mul
+    options['power_base'   ] = power_base
+    # for commutative factors: apply hint 'power_exp' of expand()
+    options['power_exp'    ] = power_exp
+    # if powers are nested, expand_mul() the product of exponents
+    if nested_exp     is None: nested_exp   = opt_mul
+    options['nested_exp'   ] = nested_exp
+    # apply lapply to exponents, otherwise exponents are used "as is"
+    options['apply_exp'    ] = apply_exp
+    # Add function, or .func per addition expression
+    options['Add'          ] = Add  # no check done
+
+    # The cache for Pow_with_Cache() is just a dictionary that is
+    # either passed as option "cache_for_pow" or provided here, so
+    # it is local to this particular invocation of lapply. This makes
+    # it threadsafe and also **bound to the values of the options**.
+    # If a dict is provided, entries will be added and the caller may
+    # thus re-use it in the next invocation.
+    options["cache_for_pow"] = cache_for_pow \
+                 if isinstance(cache_for_pow, dict) else {}
+
+    # call the workhorse
+    res = lapply_mul2([e], [], **options)
+    # clean up if any, and return
+    return res                              # End of lapply
+
+
+# some modul local constants for convenience
+EmptyMul = Mul()     # is Integer(1), is S.One, but more intuitive here
+to_the_left  = True  # option for qrec_x: split elem factor to the left
+to_the_right = False # option for qrec_x: split elem factor to the right
+
+"""
+Internal Remarks on object type constructors and Mul
+-----------------------------------------------------

```

Comment:
It's hard to read the rest of the file below this point because the syntax highlighter breaks.

Reply:
It is definitely a bug in the highlighter but it happens because you are doing something unusual and unnecessary which is to have an expression that is just a string that is not assigned to or used for anything. Comments should just be comments (lines beginning with `#`) so that the highlighter shows them as comments just like all the other comments.